src/suffer.cpp

```diff
@@ -290,6 +292,26 @@ void suffer::while_underwater( Character &you )
     }
 }
 
+void suffer::while_grabbed( Character &you )
+{
+    // get the intensity of the current grab
+    int grab_intensity = you.get_effect_int( effect_grabbed, body_part_torso );
+
+    // you should have trouble breathing as you get swarmed by zombies grabbing you
+    you.oxygen -= grab_intensity;
```

Comment:
IIRC this is number of grabs, which makes it scale too fast as at the upper extreme of attackers (8), that reaches suffocation in a mere  6 seconds or so.
The whole rate needs to be reexamined as suffocation time to unconsciousness is AFAICT several minutes,  much less death which seems to be somewhere between 5 and 15 minutes, but we can set that aside and focus on the scaling for this pr, which at a wild stab should be half the rate of being underwater with 2 attackers (impairs but does not fully prevent breathing) up to twice the rate of being underwater at 8 attackers (representing lungs being forcibly emptied by pressure).
I also think this would require or at least be enhanced by being downed, one zombie or even 2-4 zombies probably isn't *crushing* you per se while you're standing, but once downed they're going to naturally pile on and start crushing you.
While I think this is too fast in one way, it also lacks a tie-in to stamina, which would have you becoming impaired much more rapidly

Reply:
Should stamina regen be disabled while grabbed?